### PR TITLE
 add cipher profiles for external servers

### PR DESCRIPTION
**Pull request by stlaz**
updates https://github.com/openshift/api/pull/432 to current Mozilla recommendations.

OEM: https://github.com/openshift/enhancements/pull/21

@openshift/openshift-team-network-edge
@openshift/sig-master

### CODE REVIEW

**Comment by stlaz** [comment]
@danehans @deads2k @ironcladlou @knobunc 

Although `maximumVersion` is a part of the Mozilla specification, I am not sure we should include it in the API. I would expect this to be wired to `ServingInfo` which eventually uses the upstream [SecureServingOptions](https://github.com/kubernetes/apiserver/blob/master/pkg/server/options/serving.go#L37), which lacks such a field.

Is there any reason why we would like to limit the maximum TLS version used?

**Comment by ironcladlou** [comment]
Why are we duplicating https://github.com/openshift/api/pull/432?

**Comment by stlaz** [comment]
> Why are we duplicating #432?
@deads2k asked me to take it over

File: config/v1/types_tlssecurityprofile.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+			"AES128-GCM-SHA256",
+			"AES256-GCM-SHA384",
+			"AES128-SHA256",
+			"AES256-SHA256",
+			"AES128-SHA",
+			"AES256-SHA",
+			"DES-CBC3-SHA",
+		},
+		TLSVersion: TLSVersion{
+			MinimumVersion: VersionTLS10,
```
Comment (deads2k) [inline config/v1/types_tlssecurityprofile.go]: this looks questionable to me.

**Comment by deads2k** [comment]
we need something to pin the rest of the conversation. The API looks shaped correctly and we can niggle over what's included as we continue.

/lgtm

**Pull request merged**